doc/design/embargo-1.md

```diff
@@ -0,0 +1,119 @@
+give this user access.
+
+## Principles
+
+### Use of key stores
+
+Rely on the fact that "stores" (`blobs/`, and future `zarr/` see #295) although "public" are not pragmatically "usable" for users, since the assets named based on their IDs, and provide no semantic meaning (not even a file extension).
+It would be an expensive procedure for someone to "mine" the store(s) for some relevant data they would like to "steal".
+
+### "Move" operation within/between buckets is "expensive"
+
+There is no AWS S3 API for quick move (renaming) of a key within a bucket.
+As a result, "move" operation can be **very** expensive in terms of traffic (thus possibly cost if not "sponsored"), time it would take for transfer, implementation e.g. handling of possible errors during transfer, etc
+
+### Publishing a dandiset is enabled only for Open dandisets
+
+### Use-cases to fulfill
+
+- UC-Collaboration: authorized users of emabrgoed dataset should be able to manipulate it as easily as if it was a public dataset.
+- UC-Review: ability to make embargoed dataset available to other users "read-only".
+- UC-Promotion (findability): make embargoed dataset discoverable by others, so that Corresponding contact could be contacted to provide access to the dandiset.
+
+UC-Promotion is somewhat optional, this design proposal aims to un-conditional (i.e. no option to configure for it) support, but it could truly be made optional.
+
+## Implementation
+
+### Asset store(s) (`blobs/`)
+
+- No changes required per se
+- Optional: remove `s3:ListBucketVersions` and `s3:ListBucket` for the keystores prefixes. Rationale:
+    - it would totally prevent the unlikely, but possible, access to embargoed data through full navigation of the keystores
+    - it would not hinder accessibility to public (not embargoed) data since key stores are not usable *as is* anyways, access to specific keys goes through API or exported manifest files
+
+### `dandischema`
+
+- ??? move `access: List[AccessRequirements]` from `CommonModel` level to `Dandiset`, so to not requiring mutating or minting new `Asset`s records upon updating Dandisets from `embargoed` to `public`
+
+### Types of access
+
+- `open` - fully `open` from the start
+- `embargoed` - eventually auto-updates to `open` upon reaching `embargoedUntil` in `AccessRequirements`
+
+TODO: ATM [dandischema](https://github.com/dandi/dandischema/blob/master/dandischema/model_types.py) defines AccessTypes: `Open`, `Embargoed`, `Restricted`.  I am not including `Restricted` (embargoed without temporal or publish restrictions?) aside.
+
+### dandi-api
+
+#### Dandiset
+
+- Create (`POST /dandisets/`) endpoint gains optional
+  - `accessType` field to enter, defaulting to `open`
+  - `embargoedUntil` field to be specified, only if `accessType == 'embargoed'` and error out if specified for `open`
+- DB for dandiset gains `accessType` column to avoid lookup in metadata
```

Comment:
We don't actually need to include this in the design doc, since the Django model layer will take care of this for us.